subsys/net/ip/connection.c

```diff
@@ -794,7 +778,22 @@ static bool is_invalid_packet(struct net_pkt *pkt,
 		return false;
 	}
 
-	return true;
+	if (IS_ENABLED(CONFIG_NET_IPV4) && net_pkt_family(pkt) == AF_INET) {
+               if (net_ipv4_addr_cmp(&ip_hdr->ipv4->src,
+				     &ip_hdr->ipv4->dst) ||
+                   net_ipv4_is_my_addr(&ip_hdr->ipv4->src)) {
+                       my_src_addr = true;
+	       }
+	} else if (IS_ENABLED(CONFIG_NET_IPV6) &&
+		   net_pkt_family(pkt) == AF_INET6) {
+		if (net_ipv6_addr_cmp(&ip_hdr->ipv6->src,
+				      &ip_hdr->ipv6->dst) ||
+		    net_ipv6_is_my_addr(&ip_hdr->ipv6->src)) {
+			my_src_addr = true;
+		}
+	}
+
+	return my_src_addr && (src_port == dst_port);
```

Comment:
That won't work: if src_port != dst_port, this function should return true. The check on the ip address sources is only valid if and only if src_prt == dst_port.

Reply:
No, if src_port != dst_port, this function should return false.

I guess the point of moving the src_port != dst_port check to the top in 1be2706a93f1ae6 was to speed up the most common case.

Reply:
This function should check that if
* we are receiving a localhost network packet and
* that the port numbers are the same one
then that is an invalid packet. Checking just src != port at the beginning is not enough.

Reply:
was reading it the other way round, nevermind

Reply:
```
return X && Y
```
is equal to
```
if (!Y)
  return false;
return X;
```